### PR TITLE
Checkbox: retrofit DS6 changes to DS4 #225

### LINKED ISSUE #225: Checkbox: retrofit DS6 markup changes to DS4
I believe the main change was to the inline SVG version. We now always use two separate SVG symbols for checked and unchecked state instead of one specially created symbol that merged both states (and was a bit of a hack).

### PR DESCRIPTION
**Pull request by ianmcburnie**
## Description

* Bring DS6 markup changes to DS4
* Consolidate all common checkbox styles into a `base` folder
* Consolidate all common website documentation into a `common` folder

## Context

DS4 and DS6 will have parity going forward.

There are many commonalities in terms of css and documentation, so it makes sense to restructure things to reduce duplication. This can be the model going forward for other modules.

Previous source folder structure:

* ds4
    * checkbox.less
* ds6
   * checkbox.less

There was a lot of duplication across files in these folders.

New source folder structure:

* base
   * checkbox.less
* ds4
   * checkbox.less
* ds6
   * checkbox.less

Now all duplication has been removed and normalized into a base folder. The ds4 and ds6 folders now contain only the deltas (e.g. icons, colors). This does mean that we will have duplicated selectors in the dist file (hence the reason for disabling that rule in stylehint).

Previous documentation structure:

* ds4
   * checkbox.html
* ds6
   * checkbox.html

After the retrofitting, the documentation for ds4 and ds6 checkbox was essentially exactly the same. So again, to reduce duplication it has been restructured to:

* common
   * checkbox.html
* ds4
   * checkbox-custom.html
* ds6
   * checkbox-custom.html

The ds4 and ds6 file contain only the delta (because ds4 shows a different custom icon example than ds6 - we may be able to normalize this later). I added a new `ds` variable in Jekyll to control which version is included.

Dist folder structure is unchanged.

## Other

* Removed the bounce animation from DS4 checkbox (we can do fit & finish later)
* Increased size of DS4 checkbox to match DS6 (we can do fit & finish later)
* Removed the `no-duplicate-selectors` rule from stylelint for CSS (because the ds4/ds6 delta contains duplicate selectors)
* Moved some of the icon related mixins from ds6 mixins to common
* Prefixed relevant icon mixins with `ds6-` or `ds4-`
* All icon-font related mixins are deprecated going forward
* Any SVG related variables & mixins created a long time back in DS4 are deprecated going forward in favor of the new variables and mixins
* Consolidating language around "background" and "foreground", rather than "background" and "inline" (should help when it comes to high contrast related tasks)
* with this approach we can veer towards a single website that can switch between ds4 and ds6 on the fly, rather than maintaining two

## References

Closes #225

## Screenshots

It's probably best if you run this locally and test.

If we like this, I will duplicate the work for the radio module.

screen shot 2018-07-06 at 11 52 52 am

### CODE REVIEW

File: docs/test/checkbox/ds4/index.html [hunk 1]
```diff
@@ -101,10 +109,10 @@ <h2>Custom Icon</h2>
     </span>
 </div>
 
-<h2>Custom Icon Disabled</h2>
+<h2>Custom Icon, Disabled</h2>
 <div class="test">
     <span class="checkbox" style="color: #5ba71b">
-        <input aria-label="Default checkbox example" class="checkbox__custom-control" disabled type="checkbox" value="1" name="test3b" />
+        <input aria-label="Default checkbox example" class="checkbox__control" disabled checked type="checkbox" value="1" name="test3b" />
```
Comment (seangates) [inline docs/test/checkbox/ds4/index.html]: Is there a reason it became checked?

Comment (ianmcburnie) [reply docs/test/checkbox/ds4/index.html]: Good question. I just think the disabled example looks better when checked. That's all really. Let me know if I've overlooked something.

**Comment by ianmcburnie** [comment]
Also, the only real difference between the ds4 and ds6 less rules are the icons names, e.g.`ds6-` and `ds4-`. We could probably normalize even further if we wanted to, but didn't want to go that far for now.

**Comment by seangates** [comment]
I don't want to normalize too much, yet.

**Comment by ianmcburnie** [comment]
@PrashantAshok As somebody who touched the checkbox files/folders recently, do you have any thoughts or objections with this PR?

**Comment by ianmcburnie** [comment]
I just noticed that our checkbox width and height became rem based at some point (probably when we switched to background svg). However, waaaay in the past we (skin team & design team) made the decision that controls should not scale (see that textbox, button and combobox are fixed height), so in order to be consistent I'm going to set the checkbox dimensions to 18px x 18px. I get the feeling we will revisit this topic of scaling at some point in the future, but for now consistency is best.

**Comment by seangates** [comment]
Yeah, I remember (vaguely) some of those conversations.

**Pull request merged**